cli/src/cmd/watch.rs

```diff
@@ -0,0 +1,362 @@
+    if let Ok(re) = Regex::from_str(&re_str) {
+        let mut new_cmd = cmd.clone();
+        new_cmd.push("--match-path".to_string());
+        new_cmd.push(re.to_string());
+        // reconfigure the executor with a new runtime
+        let mut config = runtime.clone();
+        config.command(new_cmd);
+        // re-register the action
+        on_action(args.clone(), config, wx, cmd, has_conflicting_pattern_args, on_test);
+    } else {
+        eprintln!("failed to parse new regex {}", re_str);
+    }
+}
+
+/// Returns the env args without the `--watch` flag from the args for the Watchexec command
+fn cmd_args(num: usize) -> Vec<String> {
+    // all the forge arguments including path to forge bin
+    let mut cmd_args: Vec<_> = std::env::args().collect();
+    if let Some(pos) = cmd_args.iter().position(|arg| arg == "--watch" || arg == "-w") {
+        cmd_args.drain(pos..=(pos + num));
+    }
+    cmd_args
+}
+
+#[derive(Debug, Clone, Parser, Default)]
+pub struct WatchArgs {
+    /// File updates debounce delay
+    ///
+    /// During this time, incoming change events are accumulated and
+    /// only once the delay has passed, is an action taken. Note that
+    /// this does not mean a command will be started: if --no-restart is
+    /// given and a command is already running, the outcome of the
+    /// action will be to do nothing.
+    ///
+    /// Defaults to 50ms. Parses as decimal seconds by default, but
+    /// using an integer with the `ms` suffix may be more convenient.
+    /// When using --poll mode, you'll want a larger duration, or risk
+    /// overloading disk I/O.
+    #[clap(short = 'd', long = "delay", forbid_empty_values = true)]
+    pub delay: Option<String>,
+
+    /// Shell to use for the command, or `none` for direct execution
+    #[cfg_attr(
+        windows,
+        doc = "Defaults to Powershell. Examples: --use-shell=cmd, --use-shell=gitbash.exe"
+    )]
+    #[cfg_attr(
+        not(windows),
+        doc = "Defaults to $SHELL. Examples: --use-shell=sh, --use-shell=fish"
+    )]
+    #[clap(long = "use-shell", value_name = "shell")]
+    pub shell: Option<String>,
+
+    /// Don’t restart command while it’s still running
+    #[clap(long = "no-restart")]
+    pub no_restart: bool,
+
+    /// Show paths that changed
+    #[clap(long = "why")]
+    pub why: bool,
+
+    /// Watch specific file(s) or folder(s)
+    ///
+    /// By default, the project's source dir is watched
+    #[clap(
+        short = 'w',
+        long = "watch",
+        value_name = "path",
+        min_values = 0,
+        multiple_values = true,
+        multiple_occurrences = false
+    )]
+    pub watch: Option<Vec<PathBuf>>,
+}
+
+/// Returns the Initialisation configuration for [`Watchexec`].
+pub fn init() -> eyre::Result<InitConfig> {
+    let mut config = InitConfig::default();
+    config.on_error(SyncFnHandler::from(|data| -> std::result::Result<(), Infallible> {
+        eprintln!("[[{:?}]]", data);
```

Comment:
same, maybe tracing::trace? tracing::error will always be logged which isn't ideal either

Reply:
ah yeah these include all watchexec related errors as well, lets trace them instead